File: backend/src/main/java/todoktodok/backend/comment/application/dto/response/CommentResponse.java
package todoktodok.backend.comment.application.dto.response;

import java.time.LocalDateTime;
import todoktodok.backend.comment.domain.Comment;
import todoktodok.backend.member.application.dto.response.MemberResponse;

public record CommentResponse(
        Long commentId,
        MemberResponse member,
        LocalDateTime createdAt,
        String content,
        int likeCount,
        int replyCount
) {

    public CommentResponse(
            final Comment comment,
            final int likeCount,
            final int replyCount
    ) {
        this(
                comment.getId(),
                new MemberResponse(comment.getMember()),
                comment.getCreatedAt(),
                comment.getContent(),
                likeCount,
                replyCount
        );
    }
}

File: android/TodokTodok/app/src/main/java/com/team/todoktodok/data/network/response/comment/CommentResponse.kt
package com.team.todoktodok.data.network.response.comment

import com.team.domain.model.Comment
import com.team.todoktodok.data.core.ext.toLocalDateTime
import com.team.todoktodok.data.network.response.discussion.MemberResponse
import com.team.todoktodok.data.network.response.discussion.toDomain
import kotlinx.serialization.Serializable

@Serializable
data class CommentResponse(
    val commentId: Long,
    val content: String,
    val createdAt: String,
    val member: MemberResponse,
    val likeCount: Int,
    val replyCount: Int,
    val isLikedByMe: Boolean,
)

fun CommentResponse.toDomain() =
    Comment(
        commentId,
        content,
        member.toDomain(),
        createdAt.toLocalDateTime(),
        likeCount,
        replyCount,
        isLikedByMe,
    )
